### PR TITLE
Add docs for generating a custom lock file

### PR DESCRIPTION
**Pull request by hoodmane**
cc @Alex-Monahan

### CODE REVIEW

File: docs/usage/lock-files.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Generating a custom lock file
+
+For applications that have static dependencies that are not part of the Pyodide
```
Comment (hoodmane) [inline docs/usage/lock-files.md:3]:
```suggestion
For applications that have static dependencies that are not part of the default Pyodide
```

**Comment by hoodmane** [comment]
Advice doesn't work right now due to: https://github.com/pyodide/micropip/issues/107 but hopefully soon.

**Review by ryanking13 — APPROVED** [review]
Thanks! I am not sure if we want to promote the use of `pyodide venv` in this way, but we can certainly improve docs when we find a better way.

**Comment by hoodmane** [comment]
I think this should be a supported use case of `pyodide venv`: we provide no other way to get decent solves for the installations. But we need to actually make `micopip.freeze()` handle it somehow.

**Pull request closed without merging**